Publish 22.0.0.3 release blug (#2089)
OpenLiberty/blogs: posts/2022-03-15-sql-retries-22003.adoc

=== Allow SQL Operations to be retried in transaction recovery logs

In Liberty, transaction services are implicitly activated when you specify features that use transactions, such as  `jpa-2.0`, `jdbc-4.0`, and `wasJmsServer-1.0`.

When Liberty detects a failure to write to its transaction recovery logs, it invalidates the logs and disallows further global transactions. To resume transactional work, you have to restart the server and reload any applications, which can take considerable time in some cases.

Under certain circumstances, such as when the recovery log failure is caused by transient database connection errors, the condition can be handled by retrying the database operation thus avoiding the need to shutdown the server and reload the apps.

A new attribute is available for the `transaction` configuration element in the Liberty `server.xml` file that enables these transaction log SQL operations to be retried. It is named `enableLogRetries` and should be set to `true` to enable retries. For example,

[source,xml]
----
<transaction
dataSourceRef="tranlogDataSource"
recoverOnStartup="true"
waitForRecovery="false"
heuristicRetryInterval="10"
enableLogRetries="true"
/>
----

> By default the new functionality is disabled.

Reviewer: ```suggestion
By default, the new functionality is disabled.
```

> For more information on the storing or Transaction Recovery Logs in an `RDBMS`, please refer to link:{url-prefix}/docs/latest/transaction-service.html#log[Transaction log management] doc.

Reviewer: ```suggestion
For more information about storing transaction recovery logs in an RDBMS, see link:{url-prefix}/docs/latest/transaction-service.html#log[Transaction log management].
```
